controllers/resource_controller.go

```diff
@@ -70,6 +70,17 @@ func (r *ResourceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		return ctrl.Result{}, retErr
 	}
 
+	snapshot, err := CheckIfSnapshotExists(ctx, r.Client, obj.Spec.SnapshotTemplate.Name, obj.Namespace)
+	if err != nil {
+		retErr = errors.Join(retErr, err)
+		return ctrl.Result{}, retErr
+	}
+
+	if snapshot != nil && conditions.IsTrue(snapshot, meta.ReadyCondition) {
+		log.Info("snapshot found and is done for resource, re-queueing", "snapshot", obj.Spec.SnapshotTemplate.Name)
+		return ctrl.Result{RequeueAfter: obj.GetRequeueAfter()}, nil
+	}
```

Comment:
Would make more sense to me if this was in `r.reconcile`.

Reply:
and I don't think this is a deep-enough check. We need to check: has the source component, resource or config changed and if none of them have changed and the snapshot exists we don't need to reconcile. Otherwise we need to reconcile again.

Reply:
>and I don't think this is a deep-enough check. We need to check: has the source component, resource or config changed and if none of them have changed and the snapshot exists we don't need to reconcile. Otherwise we need to reconcile again.

I don't necessarily agree with this. That would mean that you could overwrite a snapshot with new data, wouldn't it? And generally, snapshots, once done, should pretty much be immutable, right?

Reply:
> Would make more sense to me if this was in r.reconcile.

Right, but it's used in two places, so I was reluctant to do that because I would have to copy the whole check in there as well.

Reply:
> I don't necessarily agree with this. That would mean that you could overwrite a snapshot with new data, wouldn't it? And generally, snapshots, once done, should pretty much be immutable, right?

The Snapshot Custom Resource is mutable as it is effectively just a pointer to a blob.